nns/nns_contract.go

```diff
@@ -503,6 +516,13 @@ func getRecordKey(tokenId []byte, name string, typ RecordType) []byte {
 	return append(recordKey, []byte{byte(typ)}...)
 }
 
+// addUniqueBytesToRecordKey add to key 4 unique bytes to allow store several records with the same type.
+func addUniqueBytesToRecordKey(recordKey []byte) []byte {
+	tx := runtime.GetScriptContainer()
+	bb := []byte(tx.Hash)
+	return append(recordKey, bb[:4]...)
+}
```

Comment:
I think, maintaining a counter is more reliable. This implementation also prohibits us from setting multiple records in a single transaction (which is unexpected behaviour from user POV).

Reply:
If we stay with the random numbers, we better use `runtime.GetRandom`, as it solves the problem with multiple queries in a single tx.